colserde: add Interval serialization

asubiotto opened the pull request:
This allows us to fall back to disk when using this type and send the type
over the wire.

Release note (sql change): vectorized distributed flows and disk spilling now
support the INTERVAL type.

Closes #45392

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45776)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 10 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45776#-:-M1gjyw7CDgCfkoBhjU1:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/col/colserde/arrowbatchconverter.go, line 123 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45776#-M1gZETGCUthr0GceBlh:-M1gZETGCUthr0GceBli:b-xoubwe) ([raw file](https://github.com/cockroachdb/cockroach/blob/79aa695bdca4adfbffc54bdd0b653ec9d0c8a869/pkg/col/colserde/arrowbatchconverter.go#L123)):*
> ```Go
> 
> 		if typ == coltypes.Bool || typ == coltypes.Decimal || typ == coltypes.Timestamp || typ == coltypes.Interval {
> 			// Bools, Decimals, and Timestamps are handled differently from other
> ```

nit: the comment seems a little stale, maybe we should remove it entirely?


<!-- Sent from Reviewable.io -->

asubiotto commented:
Seems like the CI failures are unrelated flakes (timeouts and `1.5s` heartbeats)

Commit:
colserde: add Interval serialization

This allows us to fall back to disk when using this type and send the type
over the wire.

Release note (sql change): vectorized distributed flows and disk spilling now
support the INTERVAL type.
